shell/platform/windows/flutter_windows_engine.cc

```diff
@@ -492,34 +494,118 @@ bool FlutterWindowsEngine::Stop() {
 
 std::unique_ptr<FlutterWindowsView> FlutterWindowsEngine::CreateView(
     std::unique_ptr<WindowBindingHandler> window) {
-  // TODO(loicsharma): Remove implicit view assumption.
-  // https://github.com/flutter/flutter/issues/142845
+  auto view_id = next_view_id_;
   auto view = std::make_unique<FlutterWindowsView>(
-      kImplicitViewId, this, std::move(window), windows_proc_table_);
+      view_id, this, std::move(window), windows_proc_table_);
 
   view->CreateRenderSurface();
 
-  views_[kImplicitViewId] = view.get();
+  next_view_id_++;
+
+  {
+    // Add the view to the embedder. This must happen before the engine
+    // is notified the view exists and starts presenting to it.
+    fml::UniqueLock write_lock{*views_mutex_};
+    FML_DCHECK(views_.find(view_id) == views_.end());
+    views_[view_id] = view.get();
+  }
+
+  if (!view->IsImplicitView()) {
+    FML_DCHECK(running());
+
+    struct Captures {
+      fml::AutoResetWaitableEvent latch;
+      bool added;
+    };
+    Captures captures;
```

Comment:
Should this struct be defined within the function, and if so, should the variable be declared separately like this?

Reply:
Yup, here's prior art in the Windows embedder:

https://github.com/flutter/engine/blob/e4e274898efc99f2d672daf774fd17a04e2d0da2/shell/platform/windows/flutter_windows_engine.cc#L752-L767

Slight difference: for adding/removing views we can stack allocate the captures, we don't need to heap allocate.